discord/bot.py

```diff
@@ -381,9 +381,10 @@ async def register_commands(
 
         commands = [copy.copy(cmd) for cmd in commands]
 
-        for cmd in commands:
-            to_rep_with = [guild_id] if guild_id is not None else guild_id
-            cmd.guild_ids = to_rep_with
+        if guild_id is not None:
```

Comment:
Nice catch, but let's use MISSING instead of None so we can still make the commands global if we want to. (Also edit the default to MISSING)
```suggestion
        if guild_id is not MISSING:
```